static/js/filter.js

```diff
@@ -46,6 +46,16 @@ function message_in_home(message) {
 
 function message_matches_search_term(message, operator, operand) {
     switch (operator) {
+    case 'has':
+        if (operand === 'image') {
+            return message.has_image;
+        } else if (operand === 'link') {
+            return message.has_link;
+        } else if (operand === 'attachment') {
+            return message.has_attachment;
+        }
+        return false; // is:whatever returns true
```

Comment:
I don't think we have to send the data from the server; fundamentally, we can just iterate through the link tags in the HTML via jQuery to do a completely client-side check for each of these.

Reply:
Ok I will try that approach.

Reply:
@timabbott I am able to detect images, link 
but how to distinguish between normal links and attachments like pdf, mp4, etc. ? 
Can i rely on checking if `/user_uploads/` exist in the href of the anchor tag ?

Reply:
Yes, `attachment` has a very specific meaning: a link to a `/user_uploads` URL.